vhost-user-backend/src/handler.rs

```diff
@@ -90,6 +91,29 @@ struct AddrMapping {
     gpa_base: u64,
 }
 
+#[cfg(feature = "postcopy")]
+impl fmt::Debug for AddrMapping {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("AddrMapping")
+            .field("local_addr", &format!("0x{:x}", &self.local_addr))
+            .field("vmm_addr", &format!("0x{:x}", &self.vmm_addr))
+            .field("size", &format!("0x{:x}", &self.size))
+            .field("gpa_base", &format!("0x{:x}", &self.gpa_base))
+            .finish()
+    }
+}
+
+#[cfg(not(feature = "postcopy"))]
+impl fmt::Debug for AddrMapping {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("AddrMapping")
+            .field("vmm_addr", &format!("0x{:x}", &self.vmm_addr))
+            .field("size", &format!("0x{:x}", &self.size))
+            .field("gpa_base", &format!("0x{:x}", &self.gpa_base))
+            .finish()
+    }
+}
+
```

Comment:
Why not just?:

```rust
#[derive(Debug)]
struct AddrMapping {
...
}
```

since all `AddrMapping`'s fields are primitive

Reply:
Addresses are not represented as hexadecimal with `#[derive(Debug)]`

Reply:
If the caller wants hex values they can do "{:x?}"

Reply:
That's news to me :) thanks for showing that. In that case `#[derive(Debug)]` is sufficient. Will make the change.